src/soilId/soilIdService.ts

```diff
@@ -17,9 +17,44 @@
 
 import { graphql } from 'terraso-client-shared/graphqlSchema';
 import { SoilIdInputData } from 'terraso-client-shared/graphqlSchema/graphql';
+import { soilDataToIdInput } from 'terraso-client-shared/soilId/soilIdFunctions';
+import { SoilData } from 'terraso-client-shared/soilId/soilIdTypes';
 import * as terrasoApi from 'terraso-client-shared/terrasoApi/api';
 import { Coords } from 'terraso-client-shared/types';
 
+export const fetchSoilMatches = async ({
+  coords,
+  siteId,
+  soilData,
+}: {
+  coords: Coords;
+  siteId?: string;
+  soilData?: SoilData;
+}) => {
+  /*
+   * Depending on the given input parameters, one of the two APIs needs to be queried.
+   * Sites with data use the "data based" soil matches API, as it can use soil data and location
+   * Temporary locations use the "location based" soil matches API, only using location
```

Comment:
Maybe something like this.
```suggestion
   * We call different APIs for temporary locations and sites.
   * Temporary locations: no soil data is sent, just location. Calls `list_soils` only.
   * Sites : both soil data and location are sent. Calls `list_soils` and `rank_soils`.
```

Reply:
Looking at the code, this isn't entirely accurate. The separation isn't "Sites vs. Temporary locations"; it's "Sites with data vs. Temporary locations and sites without data". I'm also somewhat hesitant to refer to the names of things in code in case they change, or because I see them as implementation details from the perspective of this code. So I'm gonna let @tm-ruxandra chime in here if she has thoughts.

Reply:
I've revised the comments to reflect your correction. Feel free to edit futher.

Reply:
I agree with Carissa that it's inappropriate to make reference to soil ID methods that aren't visible anywhere else in the frontend codebase. I will revise the comment for greater clarity.